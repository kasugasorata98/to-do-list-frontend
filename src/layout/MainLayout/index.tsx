import { colors } from "@/styles/colors";
import { Box } from "@chakra-ui/react";

import React, { CSSProperties, ReactNode } from "react";

const MainLayout: React.FC<{
  children: ReactNode;
  style?: CSSProperties;
}> = ({ children, style }) => {
  return (
    <Box
      height={"100%"}
      bg={colors.grey}
      display={"flex"}
      justifyContent="center"
      style={{ ...style }}
    >
      {children}
    </Box>
  );
};
export default MainLayout;
