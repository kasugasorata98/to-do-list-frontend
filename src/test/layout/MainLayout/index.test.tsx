import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import MainLayout from "@/layout/MainLayout";

describe("MainLayout", () => {
  it("matches snapshot", () => {
    const { container } = render(<MainLayout>Hello world</MainLayout>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the children", () => {
    const { getByText } = render(
      <MainLayout>
        <div>Test Children</div>
      </MainLayout>
    );

    const childElement = getByText("Test Children");
    expect(childElement).toBeInTheDocument();
  });

  it("applies the style prop", () => {
    const { getByTestId } = render(
      <MainLayout data-testid="main-layout" style={{ backgroundColor: "red" }}>
        <div>Test Children</div>
      </MainLayout>
    );

    const mainLayoutElement = getByTestId("main-layout");
    expect(mainLayoutElement).toHaveStyle("background-color: red");
  });

  it("applies style prop to Box component", () => {
    const style = {
      width: "50%",
      height: "50%",
      backgroundColor: "red",
    };
    const { container } = render(
      <MainLayout style={style} children={undefined} />
    );
    const box = container.firstChild;

    expect(box).toHaveStyle("width: 50%");
    expect(box).toHaveStyle("height: 50%");
    expect(box).toHaveStyle("background-color: red");
  });
});
