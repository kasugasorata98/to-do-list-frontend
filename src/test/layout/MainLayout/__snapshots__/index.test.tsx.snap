// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout matches snapshot 1`] = `
<div
  class="css-rogafi"
>
  Hello world
</div>
`;
